api/app/Actions/EventType/AddEventTypeAction.php

```diff
@@ -29,8 +35,10 @@ public function execute(AddEventTypeRequest $request): AddEventTypeResponse
         $eventType->timezone = $request->getTimezone();
         $eventType->disabled = $request->getDisabled();
 
-        $eventType = $this->eventTypeRepository->save($eventType);
-        $this->eventTypeRepository->saveAvailabilities($eventType, $request->getAvailabilities());
+        if ($this->availabilityService->validateAvailabilities($request->getAvailabilities())) {
```

Comment:
You need to map the array of data to `Availability` entity without saving before passing.